File: Darts/src/burlton/dartzee/code/bean/GameParamFilterPanel.java
package burlton.dartzee.code.bean;

import java.awt.BorderLayout;
import java.awt.event.ActionListener;

import javax.swing.JPanel;

public abstract class GameParamFilterPanel extends JPanel
{
	public GameParamFilterPanel()
	{
		setLayout(new BorderLayout(0, 0));
	}
	
	public abstract String getGameParams();
	public abstract void setGameParams(String gameParams);
	public abstract String getFilterDesc();
	public abstract void enableChildren(boolean enabled);
	public abstract void addActionListener(ActionListener listener);
	
	@Override
	public void setEnabled(boolean enabled)
	{
		super.setEnabled(enabled);
	
		enableChildren(enabled);
	}
}

File: src/main/kotlin/dartzee/bean/GameParamFilterPanel.kt
package dartzee.bean

import java.awt.BorderLayout
import java.awt.event.ActionListener
import javax.swing.JPanel

abstract class GameParamFilterPanel : JPanel() {
    init {
        layout = BorderLayout(0, 0)
    }

    abstract fun setGameParams(gameParams: String)

    abstract fun getGameParams(): String

    abstract fun getFilterDesc(): String

    abstract fun enableChildren(enabled: Boolean)

    abstract fun addActionListener(listener: ActionListener)

    abstract fun removeActionListener(listener: ActionListener)

    override fun setEnabled(enabled: Boolean) {
        super.setEnabled(enabled)

        enableChildren(enabled)
    }
}
